testing/ef-tests/src/macros/sanity_blocks.rs

```diff
@@ -0,0 +1,88 @@
+#[macro_export]
+macro_rules! test_sanity_blocks {
+    ($processing_fn:path) => {
+        paste::paste! {
+            #[cfg(test)]
+            #[allow(non_snake_case)]
+            mod test_sanity_blocks {
+                use super::*;
+                use ream_consensus::execution_engine::mock_engine::MockExecutionEngine;
+                use std::{fs, path::{Path, PathBuf}, sync::Arc};
+                use tokio::sync::Mutex;
+                use serde_yaml;
+
+                #[derive(Debug, serde::Deserialize)]
+                struct MetaData {
+                    blocks_count: usize,
+                    bls_setting: Option<usize>,
+                }
+
+                #[tokio::test]
+                async fn test_sanity_blocks() {
+                    let base_path = std::env::current_dir().unwrap().join("mainnet/tests/mainnet/deneb/sanity/blocks/pyspec_tests");
+
+                    let mock_engine = MockExecutionEngine {
+                        execution_valid: true,
+                    };
+
+                    for entry in std::fs::read_dir(&base_path).unwrap() {
+                        let entry = entry.unwrap();
+                        let case_dir = entry.path();
+                        if !case_dir.is_dir() {
+                            continue;
+                        }
+
+                        let case_name = case_dir.file_name().unwrap().to_str().unwrap();
+
+                        let meta: MetaData = {
+                            let meta_path = case_dir.join("meta.yaml");
+                            let content = fs::read_to_string(meta_path)
+                                .expect("Failed to read meta.yaml");
+                            serde_yaml::from_str(&content).expect("Failed to parse meta.yaml")
+                        };
+
+                        let pre_state: BeaconState = utils::read_ssz_snappy(&case_dir.join("pre.ssz_snappy")).expect("cannot find test asset (pre.ssz_snappy)");
+
+                        let validate_result = true;
+
+                        let mut state = pre_state.clone();
+                        let mut result: Result<(), String> = Ok(());
+
+                        for i in 0..meta.blocks_count {
+                            let block_path = case_dir.join(format!("blocks_{}.ssz_snappy", i));
+                            if !block_path.exists() {
+                                panic!("Test asset not found: {:?}", block_path);
+                            }
+
+                            let signed_block: SignedBeaconBlock = utils::read_ssz_snappy(&block_path)
+                                .expect(&format!("cannot find test asset (blocks_{i}.ssz_snappy)"));
+
+                            let result = state.$processing_fn(signed_block, validate_result, &mock_engine).await;
```

Comment:
```suggestion
                             result = state.$processing_fn(signed_block, validate_result, &mock_engine).await;
```

the result isn't being used